windows build issue

https://github.com/gengjiawen/dukhttp/runs/2064481139?check_suite_focus=true

```console
D:\a\dukhttp\dukhttp\src\main.c(408,22): error C2079: 'act' uses undefined struct 'sigaction' [D:\a\dukhttp\dukhttp\dukhttp.vcxproj]
D:\a\dukhttp\dukhttp\src\main.c(410,9): error C2224: left of '.sa_handler' must have struct/union type [D:\a\dukhttp\dukhttp\dukhttp.vcxproj]
D:\a\dukhttp\dukhttp\src\main.c(411,14): warning C4013: 'sigaction' undefined; assuming extern returning int [D:\a\dukhttp\dukhttp\dukhttp.vcxproj]
D:\a\dukhttp\dukhttp\src\main.c(411,22): error C2065: 'SIGPIPE': undeclared identifier [D:\a\dukhttp\dukhttp\dukhttp.vcxproj]
```

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -1,0 +1,18 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Environment Information
+        run: npx envinfo
+      - name: build
+        run: |
+          cmake .
+          cmake --build .

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,10 @@
 cmake_minimum_required(VERSION 3.4)
 project(dukhttp LANGUAGES C)
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -Os -Wall -Wextra")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -flto")
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -Os -Wall -Wextra")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -flto")
+endif()
 
 # Include deps
 add_subdirectory("deps/libuv" EXCLUDE_FROM_ALL)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,5 +15,11 @@ add_executable(dukhttp src/main.c)
 
 add_dependencies(dukhttp uv_a duktape llhttp)
 
-target_link_libraries(dukhttp uv_a duktape llhttp m)
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
+  target_link_libraries(dukhttp uv_a duktape llhttp m)
+else()
+  target_link_libraries(dukhttp uv_a duktape llhttp)
+endif()
+
+target_link_libraries(dukhttp uv_a duktape llhttp)
 target_include_directories(dukhttp PUBLIC "${PROJECT_SOURCE_DIR}/src")

--- a/src/main.c
+++ b/src/main.c
@@ -403,6 +403,7 @@ int main(int argc, char** argv) {
 
   fprintf(stderr, "Listening on http://[::]:6007\r\n");
 
+#ifndef _WIN32
   /* Ignore SIGPIPE */
   {
     struct sigaction act;
@@ -410,6 +411,7 @@ int main(int argc, char** argv) {
     act.sa_handler = SIG_IGN;
     sigaction(SIGPIPE, &act, NULL);
   }
+#endif
 
   CHECK_EQ(0, uv_run(&loop, UV_RUN_DEFAULT));
 

